mdbenchmark/plot.py

```diff
@@ -0,0 +1,190 @@
+            console.info('Data for the following hosts will be plotted: {}.', set(host_name))
+        elif host not in df['host'].tolist():
+            console.error('The host {} does not exist in your csv data. Exiting.', host)
+
+    if not host_name:
+        console.info("Plotting all hosts in csv.")
+
+    if host_name:
+        df = df[df['host'].str.contains('|'.join(host_name))]
+
+    for module in module_name:
+        if module in ['gromacs', 'namd']:
+            console.info("Plotting all modules for engine {}.", module)
+        elif module in df['module'].tolist():
+            console.info("Plotting module {}.", module)
+        elif module not in df['module'].tolist():
+            console.error("The module {} does not exist in your data. Exiting.", module)
+
+    if not module_name:
+        console.info("Plotting all modules in your input data.")
+    # this should work but we need to check before whether any of the entered
+    # names are faulty/don't exist
+    if module_name:
+        df = df[df['module'].str.contains('|'.join(module_name))]
+
+    if df.empty:
+        console.error(
+            'Your selections contained no Benchmarking Information.\n'
+            'Are you sure all your selections are correct?')
+
+    return df
+
+
+@cli.command()
+@click.option(
+    '--csv',
+    help='Name of csv file to plot.',
+    multiple=True)
+@click.option(
+    '--output-name',
+    '-o',
+    help="Name of output file.")
+@click.option(
+    '--output-type',
+    '-t',
+    help="File extension for output file.",
+    type=click.Choice(['png', 'pdf', 'svg', 'ps']),
+    show_default=True,
+    default='png')
+@click.option(
+    '--module-name',
+    '-h',
+    multiple=True,
+    help="Module name or engine name (gromacs, namd) which is plotted.")
+@click.option(
+    '--host-name',
+    '-h',
+    multiple=True,
+    help="Host name which is plotted.")
+@click.option(
+    '--gpu/--no-gpu',
+    help="Plot data for GPU runs.",
+    show_default=True,
+    default=True)
+@click.option(
+    '--cpu/--no-cpu',
+    help="Plot data for CPU runs.",
+    show_default=True,
+    default=True)
+@click.option(
+    '--plot-cores',
+    help="Plot performance per core instead of nodes.",
+    show_default=True,
+    is_flag=True)
+def plot(csv, output_name, output_type, host_name, module_name, gpu, cpu, plot_cores):
+    """Generate plots from csv files"""
+
+    if not csv:
+        raise click.BadParameter(
+       'You must specify at least one csv file.',
+        param_hint='"--csv"')
+
+    df = pd.concat([pd.read_csv(c, index_col=0) for c in csv]).dropna()
+
+    df = filter_dataframe_for_plotting(df, host_name, module_name, gpu, cpu)
+
+    fig = Figure()
+    FigureCanvas(fig)
+    ax = fig.add_subplot(111)
+    ax = plot_over_group(df, plot_cores, ax=ax)
+
+    lgd = ax.legend(loc='upper center', bbox_to_anchor=(0.5, -0.175))
+    plt.tight_layout()
+
+    if output_name is None:
```

Comment:
Nit picking here: I just noticed that the name of the plot will be different from the CSV file. We should stick with the CSV filename in this case. Otherwise we end up with these seemingly unrelated files: `2018-07-23_140304.csv` and `2018-07-23_140328.png` (note the different seconds).